### DISCUSSION TITLE
Import yaml files with a Vite setup?

### DISCUSSION THREAD

**Discussion by Dylan190774** [quasarframework/quasar#15159] [General - Components / Directives / etc]
With Webpack I used `yaml-import-loader` to import some `yaml` settings files in my Vue components.

Is there a way to import `yaml `files with Vite?

**Comment by yusufkandemir** [comment]
You can try your luck with https://github.com/Modyfi/vite-plugin-yaml
You might get more info or suggestions at [Vite's Discord server](https://chat.vitejs.dev/)

**Reply by Dylan190774** [reply]
I have tried this, but I can't find out how to install this plugin in `Quasar`, since there's no `vite.config.js` file.
I find it really annoying that Quasar only uses it's own config file and you have to do things like this with `extendViteConf` , without really telling how to do this. 

Adding 
`import ViteYaml from '@modyfi/vite-plugin-yaml';`

in `quasar.config.js` already results in an error, so I'm stuck.

**Reply by yusufkandemir** [reply]
Reading the documentation is always the wisest thing to do:
https://quasar.dev/quasar-cli-vite/handling-vite#adding-vite-plugins

**Reply by Dylan190774** [reply]
Yes, and it's not working....

```
      vitePlugins: [
        [
          "@intlify/vite-plugin-vue-i18n",
          {
            // if you want to use Vue I18n Legacy API, you need to set `compositionOnly: false`
            // compositionOnly: false,

            // you need to set i18n resource including paths !
            include: path.resolve(__dirname, "./src/i18n/**"),
          },
        ],
        [
          require("@modyfi/vite-plugin-yaml"),
          {
            /* options */
          },
        ],
      ],
```
in `quasar.config.js` gives no error when I run the app, but when I import a yaml files in my component, I get:

`BabylonModel.vue?t=1671652180261:1177 Uncaught (in promise) TypeError: Cannot destructure property 'default' of 'undefined' as it is undefined.`

**Reply by yusufkandemir** [reply]
Do not pass an empty object as the options parameter, `@modyfi/vite-plugin-yaml` doesn't handle it well. Only pass it if you are actually going to define some options. Not related to Quasar, it's just how that plugin works. This should work:
`quasar.config.js > build > vitePlugins`
```js
[ require("@modyfi/vite-plugin-yaml") ]
```

`src/assets/test.yml`:
```yml
test:
  arr:
    - 1
    - 5
```

```js
// example: usage in src/IndexPage.vue
import test from "assets/test.yml";

console.log(test);
/*
  {
    test: {
      arr: [1, 5]
    }
  }
*/
```

**Reply by Dylan190774** [reply]
Thanks for your help, I did exactly what you did, but I still get an error:
`SyntaxError: The requested module '/src/assets/test.yml?import' does not provide an export named 'default'`

I think I'll put the yaml files in `public `now, and try it with `axios `and `js-yaml`

**Reply by yusufkandemir** [reply]
```js
vitePlugins: [
  [require('@modyfi/vite-plugin-yaml').default]
]
```
and
```js
vitePlugins: [
  ['@modyfi/vite-plugin-yaml']
]
```
both work

https://stackblitz.com/edit/quasar-import-yaml?file=src/pages/IndexPage.vue,src/assets/test.yaml,quasar.config.js
